src/core/net/dns_client.cpp

```diff
@@ -199,6 +198,23 @@ void Client::SendCopy(const Message &aMessage, const Ip6::MessageInfo &aMessageI
     }
 }
 
+uint16_t Client::GenerateUniqueRandomId(void)
+{
+    uint16_t randomId;
+    otError  error;
+
+    OT_UNUSED_VARIABLE(error);
```

Comment:
'error' is always used

Reply:
This is needed, because of `OT_ASSERT` macro, which is empty in some of GitHub Actions check.

Reply:
Then maybe it would be better to create a separate PR and change empty OT_ASSERT implementation for such cases to:
`#define OT_ASSERT(cond) (void)(cond);`

This way such OT_UNUSED_VARIABLE would not be needed. Thoughts?

Reply:
I don't have a strong opinion here, it's quite common to handle such cases in a place where assert is used. I think the `(void)(cond)` won't work, we could use `OT_UNUSED_VARIABLE` macro instead (to comply with all compilers). Let me resolve this conversation, feel free to file a PR with such improvement. Thanks!

Reply:
As @LuDuda mentioned the `(void)(cond)` would not work on all tool-chains. Note that we have different toolchain specific implementation for `OT_UNUSED_VARIABLE()` macro). Not if the `OT_UNUSED_VARIBLE()` can be used in `OT_ASSERT()` since the `cond` may not be a variable itself (some of the implementations of `OT_UNUSED_VARIBLE()` rely on the input param to be a variable and they use its address).